### PR TITLE
Upgrade to Ratpack 1.7.0

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

**Comment by drmaas** [comment]
@llinder mind reviewing?

**Comment by llinder** [comment]
@drmaas besides the scope handling changes in the server tracing handler, this looks ok.  My only concern is forcing users to upgrade to Ratpack 1.7.  Besides the 1.7 HTTP client interface change, are there any other new 1.7 features that this introduces?

File: src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java [hunk 1]
```diff
@@ -59,10 +59,19 @@ public void handle(Context ctx) throws Exception {
 
     //place the Span in scope so that downstream code (e.g. Ratpack handlers
     //further on in the chain) can see the Span.
-    Tracer.SpanInScope scope = tracing.tracer().withSpanInScope(span);
+    Tracer.SpanInScope scope = null;
+    try {
+      tracing.tracer().withSpanInScope(span);
```
Comment (llinder) [inline src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java]: shouldn't this be `scope = tracing.tracer().withSpanInScope(span)`?

File: src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java [hunk 2]
```diff
@@ -59,10 +59,19 @@ public void handle(Context ctx) throws Exception {
...
+    } catch (Exception e) {
+      if (scope != null) {
+        scope.close();
```
Comment (llinder) [inline src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java]: If we close this here, the span will no longer be in scope for the rest of the request handling.

**Comment by drmaas** [comment]
@llinder a few answers - 

1. The `DefaultServerTracingHandler` stuff was my bad - the change you commented on was previously reviewed and merged, and I did a bad rebase. Do you see further issues there?
2. In hindsight we could have made `getIdleTimeout` a default method, but we didn't. As such it is a breaking change as the code stands right now. This could go into a 2.6.0 release, and there could be patches on 2.5.x if people for some reason can't upgrade to ratpack 1.7. 
3. In ratpack 1.7 there are some module deprecations, but no other breaking changes I am aware of. See https://ratpack.io/versions/1.7.0 for details.

Internally for our apps, we will need a release of this in order to use ratpack 1.7, as upgrading to the 1.7 http client interface will not work with the brave-ratpack 2.5.0 `ZipkinHttpClientImpl`. I see this as a common issue for anyone using zipking and wanting to move to ratpack 1.7.x.

File: src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java (hunk 1 above)
Comment (drmaas) [reply src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java]: Ignore these changes

File: src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java (hunk 2 above)
Comment (drmaas) [reply src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java]: Ignore these changes

**Comment by llinder** [comment]
Maybe its worth revisiting the HTTP client interceptors that were introduced in Ratpack 1.6 and just drop this client wrapper code all together?  Maintaining forked code for 1.6 and 1.7 isn't great and I would prefer to avoid it if at all possible.  I started kicking around some of this work here https://github.com/openzipkin-contrib/brave-ratpack/pull/10.  Also there is a known bug in the streaming portion of the wrapper code which would be nice to resolve as well https://github.com/openzipkin-contrib/brave-ratpack/issues/27

Before going down the route of maintaining parallel 1.6 and 1.7 implementations I would like to revisit that branch and see if we can't get the interceptor polished off instead.

**Comment by drmaas** [comment]
@llinder Cool - looks like #10 needs a rebase and conflicts resolved before it can be merged. You could add ratpack 1.7 to that PR. In this PR there were other changes required when upgrading groovy/spock to match what ratpack 1.7 uses - maybe this one can be merged after #10 

Internally the lack of an idle timeout on pooled http connections has been a huge pain point, and we are really eager to start using that.

**Comment by llinder** [comment]
@drmaas fyi... I refreshed the `HttpClient` instrumentation in a new PR https://github.com/openzipkin-contrib/brave-ratpack/pull/29.  There is still one test flake but I think everything else is basically worked out.  I haven't tested this on a 1.7 project but I don't expect there to be any issues.

**Comment by drmaas** [comment]
Closing in favor of #29

**Pull request closed without merging**